src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferEditAmount.js

```diff
@@ -17,236 +17,114 @@
-  };
+  const dispatch = useDispatch();
+  const keyWalletBalances = useRootSelector((root) => root.keyBasedAssetTransfer.availableBalances);
+  const fiatCurrency = useFiatCurrency();
+  const rates = useRates();
 
-  const handleAmountChange = (text: string, assetData: AssetData) => {
-    const { token: symbol, decimals } = assetData;
-    const balance = getBalance(availableBalances, symbol);
-    const amountFormatted = text.toString().replace(/,/g, '.');
-    const amount = parseNumber(text);
-
-    let errorMessage;
-    if (!isValidNumber(text.toString())) {
-      errorMessage = t('error.amount.invalidNumber');
-    } else if (amount > balance) {
-      errorMessage = t('error.amount.exceedBalance');
-    } else if (!isValidNumberDecimals(amount, decimals)) {
-      errorMessage = t('error.amount.shouldNotHaveDecimals');
-    }
-
-    // resets or sets new
-    setErrorMessages({ ...errorMessages, [symbol]: errorMessage });
-
-    const updated = { assetData, amount: amountFormatted || 0 };
-    setUpdatedValues({ ...updatedValues, [symbol]: updated });
-  };
+  React.useEffect(() => {
+    InteractionManager.runAfterInteractions(() => {
+      inputRef.current?.focus();
+    });
+  }, []);
 
-  const renderAsset = ({ item }) => {
-    const { assetData: { token: symbol, name, icon }, draftAmount: amount } = item;
-    const assetBalance = getBalance(availableBalances, symbol);
-    const formattedAssetBalance = formatFullAmount(assetBalance);
-    const displayAmount = updatedValues[symbol]?.amount || '';
-    const errorMessage = errorMessages[symbol];
-    const value = isEmpty(updatedValues[symbol]) ? amount?.toString() : displayAmount;
+  // Fail safe
+  if (!assetData) {
     return (
-      <ListItemWithImage
-        label={name}
-        itemImageUrl={icon}
-        fallbackToGenericToken
-        rightColumnInnerStyle={{ flex: 1, justifyContent: 'center' }}
-        customAddon={(
-          <AmountInputWrapper>
-            <AmountInput
-              onChangeText={(text) => handleAmountChange(text, item.assetData)}
-              value={value}
-              placeholder={formattedAssetBalance}
-              keyboardType="decimal-pad"
-            />
-          </AmountInputWrapper>
-        )}
-        customAddonFullWidth={errorMessage && <ErrorHolder><ErrorText>{errorMessage}</ErrorText></ErrorHolder>}
+      <ContainerWithHeader
+        headerProps={{
+          centerItems: [{ title: t('transactions.title.amountEditScreen') }],
+        }}
       />
     );
-  };
+  }
 
-  const onNextPress = () => {
-    Object.values(updatedValues).forEach((updatedValue: any) => {
-      // toggle with new amount
-      removeKeyBasedAssetToTransfer(updatedValue.assetData);
-      if (!!updatedValue?.amount && updatedValue.amount > 0) {
-        addKeyBasedAssetToTransfer(updatedValue.assetData, updatedValue.amount);
-      }
-    });
-    navigation.goBack();
-  };
+  const handleSubmit = () => {
+    if (!assetData) return;
```

Comment:
`assetData` comes as nav param, so technically it can be undefined (not passed) but only if the component is incorrectly navigated to. This check is to 1) prevent eventual crash in such case, 2) satisfy type checking.

Reply:
Thanks for the clarification!